src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -219,14 +184,11 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
     OPENVINO_ASSERT(m_configs_map.find(device_id) != m_configs_map.end(), "[GPU] compile_model: Couldn't find config for GPU with id ", device_id);
 
     ExecutionConfig config = m_configs_map.at(device_id);
-    config.set_user_property(orig_config);
-    if (model->has_rt_info("runtime_options"))
-        config.apply_rt_info(context->get_engine().get_device_info(), model->get_rt_info<ov::AnyMap>("runtime_options"), is_llm(model));
-    config.apply_user_properties(context->get_engine().get_device_info());
-
-    set_cache_info(model, config);
+    config.set_user_property(orig_config, OptionVisibility::RELEASE);
```

Comment:
Shouldn't this be `OptionVisibility::RELEASE | OptionVisibility::RELEASE_INTERNAL`? It seems currently `RELEASE_INTERNAL` options can only be set using rt_info

Reply:
For now I don't change the rules of how options may be set (we need to discuss that separately). IMO, if we allow to set release_internal option via plugin API, then it's actually not an internal property at all

Reply:
I see
Maybe we can avoid exposing them to the user requesting ov::supported_properties, but still accept them for debug purpose (even for release configuration)? Probably, most of the model-compile time options can be safely enabled (like primitive fusions control, oneDNN library enabling, queue or buffer types selection, etc)